### ISSUE TITLE
support data URLs in `load_url`

### ISSUE THREAD

**Issue by soxofaan** [Open-EO/openeo-processes#498]
`load_url` currently only supports HTTP(S) URLs:

https://github.com/Open-EO/openeo-processes/blob/965bbaebd4d5984203a0437076c85a66a72a23e0/proposals/load_url.json#L12-L19

For a use case we were brainstorming about avoiding the overhead of creating/managing external URLs (for a lot of small files) and came to the idea to load from data URLs where the data can be embedded in base64 inside the process graph, without need for external files/URLs. E.g.

```json
  "lu": {
    "process_id": "load_url",
    "arguments": {
      "url": "data:application/vnd.apache.parquet;base64encodedgeoparquethere..",
```

- additional work for a back-end wouldn't be that much
- would also be a handy feature in clients to allow embedding local data directly in the process graph

**Comment by clausmichele** [comment]
Interesting! There will be the risk to create heavy process graphs probably, but the same happens with inline geoJSON anyway. Good to have another option and happy to try to support it.

**Comment by soxofaan** [comment]
Indeed, we already had issues with users embedding huge GeoJSON constructs in their process graph, so this would not create a new problem. As a matter of fact, the textual representation of GeoJSON makes it very space-inefficient and data URLs could improve the situation because of binary encoding and compression.

But still, it could be the responsibility of the clients to put reasonable thresholds on this and warn about or forbid excessive payloads

**Comment by m-mohr** [comment]
- This needs more investigation, e.g. how is the size difference between native GeoParquet and base64 encoding.
- Is there a size limit for data uris?
  - See MDN https://developer.mozilla.org/en-US/docs/Web/URI/Schemes/data for browsers
- This shouldn't be a way to sidetrack the /file_formats. So only file formats would be allowed that are listed as inpurt file formats in that endpoint. There's no way right now to map from file format to media type in openEO (though e have an open issue for it: https://github.com/Open-EO/openeo-api/issues/476)
- As loading from a url and from a data URI are somewhat different, we may want to define it in a separate process such as load_data_uri. (Side question: Is a data URI a valid URI according to the uri format in JSON Schema?)
- ...?

**Comment by m-mohr** [comment]
Any thoughts or updates on the questions raised in https://github.com/Open-EO/openeo-processes/issues/498#issuecomment-2585924028?

**Comment by soxofaan** [comment]
> How is the size difference between native GeoParquet and base64 encoding.  Online references claim about 4/3 of the original size
> Is there a size limit for data uris? ...    It should probably a low number in megabytes. 

Yes, that's true, base64 encoding has this "expanding" property, but I don't think this is a concern to worry about at the level of the openeo-processes. This feature is intended for "small data" use cases (e.g. a single or a couple of polygons in non-EPSG4326 projection). And when users might accidentally use oversize payloads: there is enough opportunity to raise warnings/errors client side.

Also note that GeoJSON itself is actually pretty poor in terms of size efficiency (e.g. a single float value in JSON easily consumes 15 or more characters), so a base64 encoded binary format might actually even be smaller than  GeoJSON.

 > This shouldn't be a way to sidetrack the /file_formats ...

Indeed, it should be aligned with `/file_formats`, e.g. as raised in https://github.com/Open-EO/openeo-api/issues/476

> As loading from a url and from a data URI are somewhat different, we may want to define it in a separate process such as load_data_uri.

I don't see them as different enough to create a separate process . In the web dev world they seem to be used interchangeably (typically to embed images) without much trouble. 

> Is a data URI a valid URI according to the uri format in JSON Schema? ...    In principle it should be, but tooling support has to be investigated

what kind of tooling are you worried about here?


> The content-type part of data URIs is optional, but we should probably require it?! The base64 encoding is also optional, but we should probably require it?!

yes, I would also think it's best to require both
